test/shared_test/lib_battery_lifetime_lmolto_test.cpp

```diff
@@ -0,0 +1,416 @@
+                                    90., 83.33, 76.67, 70., 63.33, 56.67, 50., 43.33, 36.67, 30., 23.33, 16.67,
+                                    10., 16.67, 23.33, 30., 36.67, 43.33};
+
+    while (day < 87) {
+        for (size_t hr = 0; hr < DODs_day.size(); hr++) {
+            double prev_DOD = DODs_day[hr % 24];
+            double DOD = DODs_day[hr];
+            for (size_t min = 0; min < (size_t)(1. / dt_hr); min++) {
+                bool charge_changed = (hr == 7 || hr == 19) && min == 0;
+                if (min != 0)
+                    prev_DOD = DOD;
+                model->runLifetimeModels(idx, charge_changed, prev_DOD, DOD, 25);
+                idx ++;
+            }
+        }
+        day ++;
+    }
+
+    auto state = model->get_state();
+
+    EXPECT_EQ(state.n_cycles, 86);
+    EXPECT_NEAR(state.lmo_lto->dq_relative_cal, 0.268, 1e-3);
+    EXPECT_NEAR(state.lmo_lto->dq_relative_cyc, 0, 1e-3);
+    EXPECT_NEAR(state.q_relative, 99.732, 1e-3);
+    EXPECT_NEAR(state.day_age_of_battery, 87, 1e-3);
+
+    while (day < 870) {
+        for (size_t hr = 0; hr < DODs_day.size(); hr++) {
+            double prev_DOD = DODs_day[hr % 24];
+            double DOD = DODs_day[hr];
+            for (size_t min = 0; min < (size_t)(1. / dt_hr); min++) {
+                bool charge_changed = (hr == 7 || hr == 19) && min == 0;
+                if (min != 0)
+                    prev_DOD = DOD;
+                model->runLifetimeModels(idx, charge_changed, prev_DOD, DOD, 25);
+                idx ++;
+            }
+        }
+        day ++;
+    }
+
+    state = model->get_state();
+
+    EXPECT_EQ(state.n_cycles, 869);
+    EXPECT_NEAR(state.lmo_lto->dq_relative_cal, 1.106, 1e-3);
+    EXPECT_NEAR(state.lmo_lto->dq_relative_cyc, 0, 1e-3);
+    EXPECT_NEAR(state.q_relative, 98.894, 1e-3);
+    EXPECT_NEAR(state.day_age_of_battery, 870, 1e-3);
+}
+
+TEST_F(lib_battery_lifetime_lmolto_test, CyclingEveryTwoDays) {
+    double T = 25.15;
+    size_t day = 0;
+    while (day < 87) {
+        for (size_t i = 0; i < 48; i++) {
+            size_t idx = day * 48 + i;
+            if (i == 0)
+                model->runLifetimeModels(idx, false, 90, 90, T);
+            else if (i == 1)
+                model->runLifetimeModels(idx, true, 90, 10, T);
+            else if (i < 47)
+                model->runLifetimeModels(idx, false, 10, 10, T);
+            else
+                model->runLifetimeModels(idx, true, 10, 90, T);
+        }
+        day += 2;
+    }
+
+    auto state = model->get_state();
+
+    EXPECT_EQ(state.n_cycles, 43);
+    EXPECT_NEAR(state.lmo_lto->dq_relative_cal, 0.404, 1e-3);
+    EXPECT_NEAR(state.lmo_lto->dq_relative_cyc, 0.0637, 1e-3);
+    EXPECT_NEAR(state.q_relative, 99.532, 1e-3);
+    EXPECT_NEAR(state.day_age_of_battery, 88, 1e-3);
+}
+
+/** Test focusing on how different time steps affect the integration of a day's degradation in the NMC life model.
```

Comment:
Comment says "NMC life model"

Reply:
thanks, replaced all mention of "nmc" in the file